### PR TITLE
(DOCSP-27008): @realm/react for User API Keys page

### PR DESCRIPTION
**Pull request by mongodben**
## Pull Request Info

@realm/react for User API Keys page

### Jira

- https://jira.mongodb.org/browse/DOCSP-27008

### Staged Changes

- [User API Keys (React Native)](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27008/sdk/react-native/manage-users/manage-user-api-keys/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/react-native/manage-users/manage-user-api-keys.txt [hunk 1]
```diff
@@ -11,6 +11,20 @@ Create & Manage User API Keys - React Native SDK
...
+User API keys let you give access to a user to another service without sharing
+that users credentials. User API keys can be revoked at any time by the authenticated user.
+User API keys do not expire on their own.
+
+You can manage a user API key with the :js-sdk:`ApiKeyAuth <Realm.Auth.ApiKeyAuth.html>`
+client accessed with an authenticated user's :js-sdk:`User.apiKeys
+<Realm.User.html#apiKeys>` property.
+
+If you are using ``@realm/react``, you can access a user's ``ApiKeyAuth`` client
+with the ``useUser()`` hook in a component wrapped by ``UserProvider``.
```
Comment (krollins-mdb) [inline source/sdk/react-native/manage-users/manage-user-api-keys.txt:23]: [style question] I've been writing the provider components like `<UserProvider>` to hopefully make the distinction between them and the hooks (`useUser()`) very clear. I'm not totally committed to the convention, but we should be consistent one way or the other. What do you think?

File: source/sdk/react-native/manage-users/manage-user-api-keys.txt [hunk 2]
```diff
@@ -11,6 +11,20 @@ Create & Manage User API Keys - React Native SDK
    :depth: 2
    :class: singlecol
 
+User API keys let you give access to a user to another service without sharing
```
Comment (krollins-mdb) [inline source/sdk/react-native/manage-users/manage-user-api-keys.txt:14]: This wording isn't clear to me. Did you mean:
```suggestion
User API keys let you give a user access to another service without sharing
```
That's a bit different than what we say in the [App Services docs](https://www.mongodb.com/docs/atlas/app-services/authentication/api-key/?_ga=2.212108432.528134206.1676297375-490568303.1676297375#user-api-keys), which would be more like this:
```suggestion
User API keys allow devices or services to communicate with App Services on behalf of a user without sharing
```

File: source/sdk/react-native/manage-users/manage-user-api-keys.txt (hunk 1 above)
Comment (mongodben) [reply source/sdk/react-native/manage-users/manage-user-api-keys.txt:23]: interesting idea. definitely open to using the <Parens> for components. i'll look to see if there's a best practice here. like what are the official react docs doing for example? 

also, i think what these components/hooks are will be clearer once we have the generated reference docs.

File: source/sdk/react-native/manage-users/manage-user-api-keys.txt (hunk 2 above)
Comment (mongodben) [reply source/sdk/react-native/manage-users/manage-user-api-keys.txt:14]: thx, i like the 2nd suggested phrasing. using that.

File: source/sdk/react-native/manage-users/manage-user-api-keys.txt (hunk 1 above)
Comment (krollins-mdb) [reply source/sdk/react-native/manage-users/manage-user-api-keys.txt:23]: Ooh, good question. Looks like the React Native docs are inconsistent. The [guides](https://reactnative.dev/docs/using-a-listview) seem to not have the angle brackets. But the [component docs](https://reactnative.dev/docs/flatlist) (mostly?) do.

Comment (krollins-mdb) [reply source/sdk/react-native/manage-users/manage-user-api-keys.txt:23]: This is something we can standardize on later. We could do a pass through everything in the next phase.

Comment (mongodben) [reply source/sdk/react-native/manage-users/manage-user-api-keys.txt:23]: > Ooh, good question. Looks like the React Native docs are inconsistent. The [guides](https://reactnative.dev/docs/using-a-listview) seem to not have the angle brackets. But the [component docs](https://reactnative.dev/docs/flatlist) (mostly?) do.

what's your preference? i'll defer to you

Comment (krollins-mdb) [reply source/sdk/react-native/manage-users/manage-user-api-keys.txt:23]: Let's add the angle brackets for components. I think that helps make things clearer.

**Review by krollins-mdb — APPROVED** [review]
LGTM! We can add the angle brackets in this PR or do add them in a larger sweep later on.

**Comment by mongodben** [comment]
> LGTM! We can add the angle brackets in this PR or do add them in a larger sweep later on.

let's do a sweep later on. since there's some other stuff that needs it too. and it'll be a very easy find/replace.

will make a ticket for this.

**Pull request merged**